### PR TITLE
Alikins/bind async

### PR DESCRIPTION
**Pull request by alikins**

### CODE REVIEW

File: test/functional/connection-tests.py [hunk 1]
```diff
@@ -162,16 +182,121 @@ def test_bind_no_args(self, mock_conn, mock_drift, mock_validate):
...
+        self.assertEqual(updated_status['targetId'], self.consumer_uuid)
+        self.assertNotEqual(result['updated'], updated_status['updated'])
+
+    def _verify_job(self, job_results):
+        self.assertTrue(job_results['done'])
+        self.assertFalse(job_results['state'] == "FAILED")
+
+    # TODO: replace this with an event loop and event source
+    def _wait_for_job(self, job_id, wait_interval=2, max_attempts=6):
+        # if it's not ready in 30s, reason enough to fail the test
```
Comment (awood) [inline test/functional/connection-tests.py]: This comment confuses me a bit.  Isn't the worst-case `wait_interval` * `max_attempts`?  I think the implementation is fine, but it'd be nice to correct/remove the comment.

**Comment by alikins** [comment]
I think I'd like to rename these methods to something like 'queueBindByEntitlementPoolJob', since that is more accurate.

**Comment by alikins** [comment]
Oh, and this needs cp fixes I haven't pushed yet, otherwise async bind by product fails.

File: test/functional/connection-tests.py (hunk 1 above)
Comment (alikins) [inline test/functional/connection-tests.py]: Indeed, I changed the wait_interval without updating the comment. Updated comment with more useful info.

**Pull request closed without merging**